### PR TITLE
Appointments 

### PR DESCRIPTION
**Pull request by davidmiller**
This is implementation stage 1.

This does not add the cron job that runs the sync periodically - the initial import takes many hours, and although the second run should be much faster, suitable to 1/2 times per day, we may need an alternative strategy.

This adds a detail view if and only if the patient has appointments.

Tests are a known work in progress, pending finalising the import strategy.

This should be mergeable as is - there will be no user visible changes until the management command fetch_appointments is run.

The deployment plan would be one run that does this, and manually running the command once, waiting till it finishes, then a second release that contains a periodic sync command and cron job.

The third stage will be to use appointment booking as a way to
a) include historic TB patients in the elCID cohort
b) automatically register and load patients when they have a TB appointment booked

### CODE REVIEW

File: elcid/urls.py [hunk 1]
```diff
@@ -10,6 +10,7 @@
 from elcid import api
 
 from elcid import views
```
Comment (fredkingham) [inline elcid/urls.py:12]: What's this doing?

Comment (davidmiller) [reply elcid/urls.py:12]: It's just from the feb version  when we had the TB cliniclist view in urls.

File: plugins/appointments/loader.py [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+
+Q_GET_ALL_PATIENT_APPOINTMENTS = """
+SELECT *
+FROM VIEW_ElCid_CRS_OUTPATIENTS
+WHERE vPatient_Number = @mrn
+AND insert_date > @insert_date
+"""
+
+
+def non_date_identical(frist, second):
```
Comment (fredkingham) [inline plugins/appointments/loader.py:23]: frist the worst, second the best?

Comment (fredkingham) [reply plugins/appointments/loader.py:23]: Also `non_date_identical` meaning `identical_except_date_fields`?

File: plugins/appointments/loader.py [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+    """
+    Given a dictionary of APPOINTMENT data from the upstream
+    database, and the PATIENT for whom it concerns, decide
+    whether to save or discard it, then do so.
+    """
+    appointment_id = appointment['Appointment_ID']
+    # Find out if this appointment has been seen before
+    if Appointment.objects.filter(appointment_id=appointment_id).exists():
+
+        frist = Appointment.objects.get(appointment_id=appointment_id)
```
Comment (fredkingham) [inline plugins/appointments/loader.py:49]: more frist

File: plugins/appointments/management/commands/initial_tb_import.py [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+        results = self.api.results_for_hospital_number(mrn)
+        update_lab_tests.update_tests(patient, results)
+        print('Done')
+        return patient
+
+    def handle(self, *args, **kwargs):
+        """
+        Main entrypoint for loading all TB appointments ever.
+        """
+        raise NotImplementedError('This command has not been refactored - do not run!')
```
Comment (fredkingham) [inline plugins/appointments/management/commands/initial_tb_import.py]: do we definitely want this?

File: plugins/appointments/templates/appointments/clinic_list.html [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+{% block content %}
+  <div class="row content-offset">
+    <div class="col-md-10 col-md-offset-1">
+      <h2 >Upcoming TB Clinic Appointments</h2>
+    </div>
+  </div>
+  {% for appointment in object_list %}
+    <div class="row"p>
+      <div class="col-md-2 col-md-offset-1">
+        {{ appointment.start_datetime }}
```
Comment (fredkingham) [inline plugins/appointments/templates/appointments/clinic_list.html]: Wil this definitely format the datetime in the way we want?

File: plugins/appointments/tests/test_loader.py [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+        }
+        dict_two = {
+            'Appointment_ID'            : '1234',
+            'Appointment_Start_Datetime': '2019-01-01 12:00:00'
+        }
+        appointment = Appointment(**dict_one)
+        self.assertFalse(loader.non_date_identical(appointment, dict_two))
+
+
+class SaveOrDiscardTestCase(OpalTestCase):
```
Comment (fredkingham) [inline plugins/appointments/tests/test_loader.py:53]: this seems less than useful

File: plugins/appointments/tests/test_loader.py [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+
+    def test_save_appointment(self):
+        appointment_data = {
+
+        }
+        pass
+
+
+class LoadAppointmentTestCase(OpalTestCase):
+
```
Comment (fredkingham) [inline plugins/appointments/tests/test_loader.py:67]: this seems less than useful

**Pull request merged**